### ISSUE TITLE
Timestamps of Dribbles and Ball Receivals in Opta Data

### ISSUE THREAD

**Issue by znstrider** [ML-KULeuven/socceraction#51]
Socceraction creates the timestamps for dribbles and receptions (atomic) as the midpoint (t2 - t1) / 2 between the first and subsequent actions.

The downside of that might be that you can have a short pass, then a long dribble, but the timestamp would still be in the middle (of the dribble).
The other way around might happen too with a long ball traveling a few sec that is immediately passed on, thus the timestamp being in the middle of the pass.

Would it make sense to assume a fixed passing speed and create the timestamp for the ball receival based on the distance and thus the time the ball would take from origin to destination?

Now I am not sure if this is very relevant. It might be useful for looking at holding times of the ball in those individual ball possessions for example.
Are Opta Timestamps always just in seconds? Maybe the lacking accuracy of those would make this approach unnecessary.

**Comment by probberechts** [comment]
I agree that it would be better to do something physics-based to determine the timing of the actions that are added in atomic-SPADL, but taking into account the time resolution of event stream data, it will not make a big difference overall. 

Also, I think we should move towards provider-specific converters for the atomic-SPADL format that starts from the original event stream data instead of the regular SPADL representation. E.g., for StatsBomb data this would allow using the Receival events to determine an accurate timing.
